lib/engine/game/g_18_ireland/step/special_track.rb

```diff
@@ -1,30 +1,56 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/special_track'
+require_relative 'narrow_track'
 
 module Engine
   module Game
     module G18Ireland
       module Step
         class SpecialTrack < Engine::Step::SpecialTrack
+          include NarrowTrack
           def potential_tiles(entity, _hex)
             return super unless entity.id == 'TIM'
             return [] unless (tile_ability = abilities(entity))
 
             tile_ability.tiles.map { |name| @game.tiles.find { |t| t.name == name } }
           end
 
+          def get_tile_lay(entity)
+            entity = entity.owner if entity.company?
+            super
+          end
+
+          def process_lay_tile(action)
+            return lay_tile_action(action, spender: action.entity.owner) if action.entity.id == 'BoW'
+
+            super
+          end
+
+          def available_hex(entity, hex)
+            return base_available_hex(entity.owner, hex) if entity.id == 'BoW'
+
+            super
+          end
+
           def hex_neighbors(entity, hex)
+            operator = entity.owner.corporation? ? entity.owner : entity
             if entity.id == 'DR'
               return unless (ability = abilities(entity))
               return if ability.count == 2 && hex.id != 'F4'
+            elsif entity.id == 'BoW' || entity.corporation?
+              return @game.graph.connected_hexes(operator)[hex] || @game.narrow_connected_hexes(operator)[hex]
             end
-            super
-          end
 
-          def process_lay_tile(action)
-            super
-            @game.clear_narrow_graph
+            return unless (ability = abilities(entity))
+            return if !ability.hexes&.empty? && !ability.hexes&.include?(hex.id)
+
+            if ability.type == :tile_lay && ability.reachable && !(@game.graph.connected_hexes(operator)[hex] ||
```

Comment:
I notice the `reachable` check here, but none of the 18Ireland entities definitions use `reachable: true` (this comment might be a little redundant with https://github.com/tobymao/18xx/pull/5016/files#r615523144)